cotainr/container.py

```diff
@@ -199,39 +264,103 @@ def run_command_in_container(self, *, cmd):
+        Add a verbosity level to Singularity commands.
 
-        The following metadata is added to the container sandbox:
-          - "cotainr.command": The full command line used to build the container.
-          - "cotainr.version": The version of cotainr used to build the container.
-          - "cotainr.url": The cotainr project url.
+        A mapping of the internal cotainr verbosity level to `Singularity
+        verbosity flags
+        <https://apptainer.org/docs/user/main/cli/apptainer.html#options>`_.
 
-        The container metadata may be inspected by running `singularity inspect` on the
-        built container image file.
+        Parameters
+        ----------
+        args : list
+            The list of command line arguments constituting the full
+            singularity command.
 
-        Notes
-        -----
-        The metadata entries are added to the `.singularity.d/labels.json
-        <https://apptainer.org/docs/user/main/environment_and_metadata.html#singularity-d-directory>`_
-        file.
+        Returns
+        -------
+        args : list
+            The updated list of singularity command line arguments.
+        """
+        if self._verbosity < 0:
+            # --silent (-s)
+            args.insert(1, "-s")
+        elif self._verbosity == 0:
+            # --quiet (-q)
+            args.insert(1, "-q")
+        elif self._verbosity >= 3:
+            # Assume --verbose (-v) is a debug level
+            args.insert(1, "-v")
+
+        return args
+
+    def _subprocess_runner(self, *, custom_log_dispatcher=None, args, **kwargs):
+        """
+        Wrap the choice of subprocess runner.
+
+        Parameters
+        ----------
+        custom_log_dispatcher : :class:`~cotainr.tracing.LogDispatcher`, optional
+            The custom log dispatcher to use when running the command (the
+            default is None which implies that the `SingularitySandbox` log
+            dispatcher is used).
+        args : list or str
+            Subprocess arguments.
+        kwargs : dict
+            Keyword arguments passed to the subprocess runner.
 
+        Returns
+        -------
+        process : :class:`subprocess.CompletedProcess`
+            Information about the process that ran in the container sandbox.
         """
-        labels_path = self.sandbox_dir / ".singularity.d/labels.json"
-        with open(labels_path, "r+") as f:
-            metadata = json.load(f)
-            metadata["cotainr.command"] = " ".join(sys.argv)
-            metadata["cotainr.version"] = _cotainr_version
-            metadata["cotainr.url"] = "https://github.com/DeiC-HPC/cotainr"
-            f.seek(0)
-            json.dump(metadata, f)
+        if custom_log_dispatcher is not None:
+            # When the command to be run in the container provides its own
+            # log_dispatcher
+            with custom_log_dispatcher.prefix_stderr_name(
+                prefix=self.__class__.__name__
```

Comment:
for a future change should we create better names for the outb`put than use the class name?

Reply:
Sounds reasonable. I wasn't able to come up with better short names than the class names, though.